### PR TITLE
Split server client topics

### LINKED ISSUE #2604: Refactor Zowe docs to clarify client-side and server-side topics
- Server-side VS Client-side topics are not well organized. For example, under Setup and Use, there are no high level distinction to immediately identify which are server-side topics and which are client-side topics. Should keep consistent with what’s presented on Zowe.org download section.  
image
- What is “Advanced Configuration”? A bit confusing and needs more explanation at the beginning.  
- “Extend” should be “Develop” for clarify. 

An example proposed navigation for Setup: 
- Overview
- Installing Zowe server-side components
   - Installing Zowe servers on z/OS
      - From a convenience build
      - From an SMP/E build
      - From Zowe Portable Sopftware Instance
   - Installing Zowe containers 
- Installing Zowe client-side components
   - Installing Zowe CLI
   - Installing Zowe Explorer
   - Installing Zowe Intellij plug-in
   - Installing Zowe Chat (Technical Preview)

### PR DESCRIPTION
**Pull request by samanthasusu**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../docs/contribute/contributing.md) to this repository.

- [ ] If the changes in this PR is part of the next future release, make this pull request against the **docs-staging** branch which will be published at the next release boundary. If the changes in this PR are part of the current release, use the default base branch, **master**. For more information about branches, see https://github.com/zowe/docs-site/tree/master#understanding-the-doc-branches. 
      
- [x] If this PR relates to GitHub issues in `docs-site` or other repositories, please list in Description, prefixed with **close**, **fix** or **resolve** keywords.

### Description (including links to related git issues)
Please describe your pull request.
- Reorganized the doc site sidebar according to server side and client side
- Updated the instructions accordingly
Fix #2604 
:heart:Thank you!

### CODE REVIEW

**Review by janan07 — APPROVED** [review]
Very practical PR for to improve user experience.

File: docs/user-guide/install-overview.md [hunk 1]
```diff
@@ -1,11 +1,14 @@
 # Overview
 
 The installation of Zowe&trade; consists of the following processes:
-- installation of the Zowe server components.
+
+- installation of the Zowe server-side components.
```
Comment (anaxceron) [inline docs/user-guide/install-overview.md:6]: Both "installation" bullets should start w/ a capital letter: "Installation of the Zowe ..."

File: docs/user-guide/zowe-getting-started-tutorial.md [hunk 2]
```diff
@@ -2,20 +2,23 @@
 
 Learn how to start exploring the Zowe components, applications and plug-ins.
 
-**Use core components:**
+## Use Zowe server-side components
```
Comment (anaxceron) [inline docs/user-guide/zowe-getting-started-tutorial.md:5]: I'd suggest removing all instances of "Use" in the headers b/c most of the bullets below already start w/ "Using," so it feels redundant to have the same verb both in the header and the bullets.

File: docs/user-guide/zowe-getting-started-tutorial.md [hunk 3]
```diff
@@ -2,20 +2,23 @@
...
+### Explore available plug-ins
 
 - [Zowe CLI plug-ins](cli-extending.md)
 - [Zowe Explorer extensions](ze-using-zowe-explorer-cics-ext.md)
 - [Using Zowe IntelliJ Plug-in](intellij-using.md)
 
-**Use incubator components:** 
+### Use incubator components
 
 - [Using Zowe Chat (Incubator)](zowe-chat/chat_use_interact_methods.md)
```
Comment (anaxceron) [inline docs/user-guide/zowe-getting-started-tutorial.md:24]: I would use a lowercase for "Incubator" as it is not a proper noun

**Review by anaxceron — APPROVED** [review]
Really like how streamlined it all looks now, well done!

I left some comments w/ suggestions on some minor points.

Also wanted to ask: Do we already provide somewhere in Zowe Docs definitions for "server side" and "client side"? If not, maybe it would be something you'd like to add.

**Comment by samanthasusu** [comment]
> Really like how streamlined it all looks now, well done!
> 
> I left some comments w/ suggestions on some minor points.
> 
> Also wanted to ask: Do we already provide somewhere in Zowe Docs definitions for "server side" and "client side"? If not, maybe it would be something you'd like to add.

Really good point. We have an overview for the "Getting started" topic and list the components belonging to server side and client side separately. Also the download section displays the components in the same way. 
>    
- Installation Zowe server-side components.
You can install the components either on z/OS only or you can install the components both on z/OS and on Docker.
- Installation of Zowe client-side components.
You can install Zowe CLI or Zowe Explorer, a Visual Studio Code extension powered by Zowe CLI.

**Pull request merged**